packages/google_mobile_maps/android/src/main/java/io/flutter/plugins/googlemobilemaps/GoogleMobileMapsPlugin.java

```diff
@@ -373,16 +433,57 @@ void hideOverlay() {
     parent.addView(mapView, 0);
   }
 
-  void moveCamera(final CameraUpdate cameraUpdate) {
+  void moveCamera(CameraUpdate cameraUpdate) {
     googleMap.moveCamera(cameraUpdate);
   }
 
-  void animateCamera(final CameraUpdate cameraUpdate) {
+  void animateCamera(CameraUpdate cameraUpdate) {
     googleMap.animateCamera(cameraUpdate);
   }
 
-  void addMarker(final MarkerOptions options) {
-    googleMap.addMarker(options);
+  String addMarker(MarkerOptions options) {
+    final Marker marker = googleMap.addMarker(options);
+    markers.put(marker.getId(), marker);
+    return marker.getId();
+  }
+
+  void removeMarker(String markerId) {
+    final Marker marker = marker(markerId);
+    marker.remove();
```

Comment:
Do you also want to remove from `markers`?

Reply:
Yes, good catch. Done.